static/updatenotification.sh

```diff
@@ -19,32 +19,58 @@ debug_mode
 # Check if root
 root_check
 
-REPORTEDNCVERSION=""
+NCMIN=$(curl -s -m 900 $NCREPO/ | sed --silent 's/.*href="nextcloud-\([^"]\+\).zip.asc".*/\1/p' | sort --version-sort | grep "${CURRENTVERSION%%.*}" | tail -1)
+REPORTEDMAJ=""
+REPORTEDMIN=""
 
-if [ "$CURRENTVERSION" == "$NCVERSION" ]
+if [ "$CURRENTVERSION" == "$NCVERSION" ] && [ "$CURRENTVERSION" == "$NCMIN" ]
 then
     print_text_in_color "$IGreen" "You already run the latest version! ($NCVERSION)"
     exit
 fi
 
-if [ "$REPORTEDNCVERSION" == "$NCVERSION" ]
+if [ "$REPORTEDMAJ" == "$NCVERSION" ] && [ "$REPORTEDMIN" == "$NCMIN" ]
 then
-    print_text_in_color "$ICyan" "The notification regarding the new Nextcloud update has been already reported! ($NCVERSION)"
+    print_text_in_color "$ICyan" "The notification regarding the new Nextcloud update has been already reported!"
     exit
 fi
 
-if version_gt "$NCVERSION" "$CURRENTVERSION"
+if [ "$NCVERSION" == "$NCMIN" ] && version_gt "$NCMIN" "$REPORTEDMIN" && version_gt "$NCMIN" "$CURRENTVERSION"
 then
-    if crontab -l -u root | grep $SCRIPTS/update.sh
+    sed -i "s|^REPORTEDMAJ.*|REPORTEDMAJ=$NCVERSION|" $SCRIPTS/updatenotification.sh
+    sed -i "s|^REPORTEDMIN.*|REPORTEDMIN=$NCMIN|" $SCRIPTS/updatenotification.sh
+    if crontab -l -u root | grep -q $SCRIPTS/update.sh
     then
         notify_admin_gui \
-        "New Nextcloud version!" \
-        "Nextcloud $NCVERSION just became available. Since you are running Automatic Updates at $AUT_UPDATES_TIME:00, you don't need to bother about updating the server manually, as that's already taken care of."
-        sed -i "s|^REPORTEDNCVERSION.*|REPORTEDNCVERSION=$NCVERSION|" $SCRIPTS/updatenotification.sh
+        "New minor Nextcloud Update!" \
+        "Nextcloud $NCMIN just became available. Since you are running Automatic Updates on Saturdays at $AUT_UPDATES_TIME:00, you don't need to bother about updating the server to minor Nextcloud versions manually, as that's already taken care of."
     else
         notify_admin_gui \
-        "Update available!" \
-        "Nextcloud $NCVERSION is available. Please run 'sudo bash /var/scripts/update.sh' from your CLI to update your server."
-        sed -i "s|^REPORTEDNCVERSION.*|REPORTEDNCVERSION=$NCVERSION|" $SCRIPTS/updatenotification.sh
+        "New minor Nextcloud Update!" \
+        "Nextcloud $NCMIN just became available. Please run 'sudo bash /var/scripts/update.sh minor' from your CLI to update your server to Nextcloud $NCMIN."
     fi
+    exit
+fi
+
+if version_gt "$NCMIN" "$REPORTEDMIN" && version_gt "$NCMIN" "$CURRENTVERSION"
```

Comment:
Why check ncmin again here nad in if started in line 38?

Reply:
Because the check in line 48 covers the case that ncversion and ncmin are the same which is possible. And in this case we only send one notification. 

This check here covers ncmin to check if there is a new minor update available and sends a notification for that.

Later the check for ncversion covers the case that there is a new major update available and also sends a notification in that case. 

So the idea is that if ncmin and ncversion are the same, you get only one notification. If they are not the same (e.g. there is a new minor and major update available) you get two notifications.

But if only e.g. a major update is available you get only one notification and if only a minor update is available you get also only one notification.